src/vmm/src/builder.rs

```diff
@@ -730,7 +730,7 @@ fn attach_virtio_device<T: 'static + VirtioDevice + Subscriber>(
     // The device mutex mustn't be locked here otherwise it will deadlock.
     let device = MmioTransport::new(vmm.guest_memory().clone(), device);
     vmm.mmio_device_manager
-        .register_new_virtio_mmio_device(vmm.vm.fd(), id, device, cmdline)
+        .register_mmio_virtio_add_to_cmdline(vmm.vm.fd(), id, device, cmdline)
```

Comment:
The _new_ in the name is there to say that this is a previously never registered `new` device.

Reply:
@acatangiu thanks for the comment! Here I took the function "register_new_mmio_rtc" as reference, thought that "new" refers to the fact that a device is created in the function. Is the fact that this is a previously never registered new device is already implied by "register"? In another word, do we need to register a device that's previously registered?

Reply:
Like I said, naming is hard, and current names are indeed confusing.

> Is the fact that this is a previously never registered new device is already implied by "register"? In another word, do we need to register a device that's previously registered?

On the snapshots path, we register a device that has previously been registered in the original microVM where the snapshot comes from. The difference being that there are two paths:
 1. the vm is fresh booted and a (new) device is registered on the first free slot
 2. the vm is a restore and a device is registered at a specific slot (same address where it was in the original/snapshot-source vm)

Alternative naming ideas that cross my mind right now:
 - `register_mmio_virtio_add_to_cmdline` for the boot and `register_mmio_virtio_at_slot` for snapshot
  or simply
 - `register_mmio_virtio_for_boot` and `register_mmio_virtio_from_snapshot`

Reply:
@acatangiu sounds good to me! Please correct me if I'm wrong, I think the restore path you are referring to is in src/device_manager/persist::restore, which call `register_virtio_mmio_device` directly. Should we just rename `register_new_virtio_mmio_device` to `register_mmio_virtio_for_boot`, without changing `register_virtio_mmio_device` to `register_mmio_virtio_from_snapshot`? This is because the second function is also called internally from the first one.